src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/KBestHaplotypeFinder.java

```diff
@@ -376,10 +371,8 @@ public void printDOT(final File file) throws FileNotFoundException {
         Utils.nonNull(file, "the output file cannot be null");
         final PrintWriter out = new PrintWriter(file);
         printDOT(out);
-        if (out.checkError()) {
-            throw new IllegalStateException("error occurred while writing k-best haplotype search graph into file '"
+        Utils.validate(!out.checkError(), () -> "error occurred while writing k-best haplotype search graph into file '"
```

Comment:
Whenever the error message is not a literal, the version with a `Supplier<String>` spares you from constructing the message when the error is not thrown.  It's not important here but I feel it's good practice to use `() ->` for any non-literal message.